### PR TITLE
WIP: Support validation of loadBalancer IPs

### PR DESCRIPTION
**Pull request by justinsb**
We also want to protect against loadBalancer IPs, so we create a
second allowlist of CIDRs for these.

### CODE REVIEW

**Comment by justinsb** [comment]
Marking this as WIP so we can explore the problem space.

* Do we need to protect against IP injection here? (I think we should offer it)
* Do we want to look at the user that issued the request, to restrict e.g. to k-c-m?
* What's the right flag syntax to use here?  Should we start using CRDs?

**Comment by SaranBalaji90** [comment]
@justinsb Good idea!! But I have few questions

1. Should we restrict based on the IP or should we restrict based on the Users for LB IPs(mostly for the status fields)? For example, in AWS CloudProvider we won't know the IP address allocated by the AWS Services until the resources are created. If its not the case with most of the other CloudProvider and IP address will be known in advance it will be a good feature to have.
2. If we are supporting both user based restriction as well as CIDRs then CRD would make it easier for users to configure. But I don't think we want to keep this webhook to prevent such attack in the long run.

File: main.go [hunk 1]
```diff
@@ -44,13 +44,15 @@ func init() {
...
+	var allowedExternalIPCIDRs []string
+	var allowedLoadBalancerIPCIDRs []string
 	var metricsAddr string
 	var webhookPort int
 
 	flag.IntVar(&webhookPort, "webhook-port", 9443, "Webhook port number")
 	flag.StringVar(&metricsAddr, "metrics-addr", "0", "The address the metric endpoint binds to.")
-	flag.StringSliceVar(&allowedCIDRs, "allowed-external-ip-cidrs", []string{}, "List of CIDR ranges allowed as External IPs in the service spec.")
+	flag.StringSliceVar(&allowedExternalIPCIDRs, "allowed-external-ip-cidrs", []string{}, "List of CIDR ranges allowed as External IPs in the service spec.")
+	flag.StringSliceVar(&allowedLoadBalancerIPCIDRs, "allowed-loadbalancer-ip-cidrs", []string{"0.0.0.0/0"}, "List of CIDR ranges allowed as External IPs in the service spec.")
```
Comment (tallclair) [inline main.go:55]:
```suggestion
	flag.StringSliceVar(&allowedLoadBalancerIPCIDRs, "allowed-loadbalancer-ip-cidrs", []string{"0.0.0.0/0"}, "List of CIDR ranges allowed as Loadbalancer Ingress IPs in the service status.")
```

**Review by tallclair — COMMENTED** [review]
Sorry, just realized I had unpublished comments from months ago...

**Pull request closed without merging**